crates/wasmtime/src/module.rs

```diff
@@ -493,23 +477,29 @@ impl Module {
         engine: &Engine,
         mmap: MmapVec,
         info: Option<CompiledModuleInfo>,
-        types: Arc<TypeTables>,
+        types: TypeTables,
     ) -> Result<Self> {
-        let module = CompiledModule::from_artifacts(
+        let module = Arc::new(CompiledModule::from_artifacts(
```

Comment:
I ideally wanted to remove this `Arc` as well but the `GlobalModuleRegistry` needed to hang on to something and the `Module` isn't constructed yet (and `Module` needs the dtor to unregister), so I went ahead and left this in an `Arc` but at this `Module` layer instead of the previous layer.